python/graphstorm/dataloading/dataset.py

```diff
@@ -117,17 +130,34 @@ class GSgnnData():
     edge_feat_field : str or dict of str
         The field of the edge features. It's a dict if different edge types have
         different feature names.
+    use_wholegraph: bool
+        Whether to use wholegraph for feature transfer.
     """
 
-    def __init__(self, graph_name, part_config, node_feat_field, edge_feat_field):
+    def __init__(self, graph_name, part_config, node_feat_field, edge_feat_field, use_wholegraph):
         self._g = dgl.distributed.DistGraph(graph_name, part_config=part_config)
+
         self._node_feat_field = node_feat_field
         self._edge_feat_field = edge_feat_field
 
         self._train_idxs = {}
         self._val_idxs = {}
         self._test_idxs = {}
 
+        if use_wholegraph and is_distributed():
+            self._wg_init = False
+            num_parts = self._g.get_partition_book().num_partitions()
+            for ntype in node_feat_field.keys():
```

Comment:
Can we store edge feature in whole graph?

Reply:
Not at this moment due to our tight deadline.